packages/plugins/response-cache/src/index.ts

```diff
@@ -20,13 +20,13 @@ export type BuildResponseCacheKeyFunction = (
   },
 ) => ReturnType<EnvelopBuildResponseCacheKeyFunction>;
 
-export type UseResponseCacheParameter = Omit<
+export type UseResponseCacheParameter<TServerContext> = Omit<
   UseEnvelopResponseCacheParameter,
   'getDocumentString' | 'session' | 'cache' | 'enabled' | 'buildResponseCacheKey'
 > & {
   cache?: Cache;
-  session: (request: Request) => PromiseOrValue<Maybe<string>>;
-  enabled?: (request: Request) => boolean;
+  session: (request: Request, serverContext?: TServerContext) => PromiseOrValue<Maybe<string>>;
+  enabled?: (request: Request, serverContext?: TServerContext) => boolean;
```

Comment:
Why is `serverContext` optional? Can conditional optional based on the provided `TServerContext` type instead?

Reply:
I don't know, the server context object passed here is `TServerContext | undefined`;
https://github.com/dotansimha/graphql-yoga/pull/3285/files#diff-2bb04768b60ebe5776d0a80eafa2b96026b7bae21deda57af2e04bb9e5abb4cbR445
It doesn't work otherwise.

Reply:
if it is `| undefined`, `serverContext: TServerContext` should be fine?

Reply:
because the `?` is the same as if it is `undefined`?

Reply:
In what scenario is the server context undefined in real-world usage?